src/ape/api/accounts.py

```diff
@@ -215,6 +221,9 @@ def deploy(
             :class:`~ape.contracts.ContractInstance`: An instance of the deployed contract.
         """
         txn = contract(*args, **kwargs)
```

Comment:
I believe that this line actually front runs the below error with the one in `ape.contracts.base`

Try removing the next line and see if your test still passes

Reply:
![image](https://github.com/ApeWorX/ape/assets/31238298/d4b3c621-63a9-4c6c-a03d-2d004a6e63fa)
Both `account.deploy` & `project.contract.deploy` test is working. But if I don't put  

```py
if kwargs.get("value") and not contract.contract_type.constructor.is_payable:
            raise MethodNonPayableError("Sending funds to a non-payable constructor.")
```
in line `224` in `src/ape/api/accounts.py` the `test_sending_funds_to_non_payable_constructor_by_accountDeploy` test will fail like you said it's being front run by ` txn = contract(*args, **kwargs)` . I tried to catch the exception here `src/ape/contracts/base.py#L1347` also but for some reason, it was not catching, instead it moved to the next exception i.e. `ContractLogicError`.